metricbeat/docker-compose.yml

```diff
@@ -1,132 +1,83 @@
-
-  # This is a proxy used to block beats until all services are healthy.
-  # See: https://github.com/docker/compose/issues/4369
-  proxy_dep1:
-    image: busybox
-    depends_on:
-      aerospike:      { condition: service_healthy }
-      apache:         { condition: service_healthy }
-      ceph:           { condition: service_healthy }
-      couchbase:      { condition: service_healthy }
-      dropwizard:     { condition: service_healthy }
-      elasticsearch:  { condition: service_healthy }
-      haproxy:        { condition: service_healthy }
-      http:           { condition: service_healthy }
-
-
-  proxy_dep2:
-    image: busybox
-    depends_on:
-      jolokia:        { condition: service_healthy }
-      kafka:          { condition: service_healthy }
-      kibana:         { condition: service_healthy }
-      kubernetes:     { condition: service_healthy }
-      kube-state:     { condition: service_started }
-      memcached:      { condition: service_healthy }
-      mongodb:        { condition: service_healthy }
-
-  proxy_dep3:
-    image: busybox
-    depends_on:
-      mysql:          { condition: service_healthy }
-      nginx:          { condition: service_healthy }
-      phpfpm:         { condition: service_healthy }
-      postgresql:     { condition: service_healthy }
-      prometheus:     { condition: service_healthy }
-      rabbitmq:       { condition: service_healthy }
-      redis:          { condition: service_healthy }
-      zookeeper:      { condition: service_healthy }
+      - ./module/aerospike/_meta/env
+      - ./module/apache/_meta/env
+      - ./module/ceph/_meta/env
+      - ./module/couchbase/_meta/env
+      - ./module/dropwizard/_meta/env
+      - ./module/elasticsearch/_meta/env
+      - ./module/haproxy/_meta/env
+      - ./module/http/_meta/env
+      - ./module/jolokia/_meta/env
+      - ./module/kafka/_meta/env
+      - ./module/kibana/_meta/env
+      - ./module/kubernetes/_meta/env
+      - ./module/memcached/_meta/env
+      - ./module/mongodb/_meta/env
+      - ./module/mysql/_meta/env
+      - ./module/nginx/_meta/env
+      - ./module/php_fpm/_meta/env
+      - ./module/postgresql/_meta/env
+      - ./module/prometheus/_meta/env
+      - ./module/rabbitmq/_meta/env
+      - ./module/redis/_meta/env
+      - ./module/zookeeper/_meta/env
 
   # Modules
   aerospike:
-    build: ${PWD}/module/aerospike/_meta
+    build: ./module/aerospike/_meta
 
   apache:
-    build: ${PWD}/module/apache/_meta
+    build: ./module/apache/_meta
 
   ceph:
-    build: ${PWD}/module/ceph/_meta
+    build: ./module/ceph/_meta
 
   couchbase:
-    build: ${PWD}/module/couchbase/_meta
+    build: ./module/couchbase/_meta
 
   dropwizard:
-    build: ${PWD}/module/dropwizard/_meta
+    build: ./module/dropwizard/_meta
 
   elasticsearch:
-    extends:
-      file: ${ES_BEATS}/testing/environments/${TESTING_ENVIRONMENT}.yml
-      service: elasticsearch
+    build: ./module/elasticsearch/_meta
```

Comment:
Not critical but the nice thing about having only one place where ES was defined is that there was only 1 place to have it update and all the special configs below.

Reply:
I know :(, basically `libcompose` doesn't support that syntax, it seems it has been deprecated for version 3 format too

Reply:
yes, which is the reason we are still stuck with 2.1 👎